packages/mermaid/src/diagrams/xychart/chartBuilder/Orchestrator.ts

```diff
@@ -0,0 +1,202 @@
+    private chartConfig: XYChartConfig,
+    private chartData: XYChartData,
+    chartThemeConfig: XYChartThemeConfig,
+    tmpSVGGElem: SVGGType
+  ) {
+    this.componentStore = {
+      title: getChartTitleComponent(chartConfig, chartData, chartThemeConfig, tmpSVGGElem),
+      plot: getPlotComponent(chartConfig, chartData, chartThemeConfig),
+      xAxis: getAxis(
+        chartData.xAxis,
+        chartConfig.xAxis,
+        {
+          titleColor: chartThemeConfig.xAxisTitleColor,
+          labelColor: chartThemeConfig.xAxisLabelColor,
+          tickColor: chartThemeConfig.xAxisTickColor,
+          axisLineColor: chartThemeConfig.xAxisLineColor,
+        },
+        tmpSVGGElem
+      ),
+      yAxis: getAxis(
+        chartData.yAxis,
+        chartConfig.yAxis,
+        {
+          titleColor: chartThemeConfig.yAxisTitleColor,
+          labelColor: chartThemeConfig.yAxisLabelColor,
+          tickColor: chartThemeConfig.yAxisTickColor,
+          axisLineColor: chartThemeConfig.yAxisLineColor,
+        },
+        tmpSVGGElem
+      ),
+    };
+  }
+
+  private calculateVerticalSpace() {
+    let availableWidth = this.chartConfig.width;
+    let availableHeight = this.chartConfig.height;
+    let plotX = 0;
+    let plotY = 0;
+    let chartWidth = Math.floor((availableWidth * this.chartConfig.plotReservedSpacePercent) / 100);
+    let chartHeight = Math.floor(
+      (availableHeight * this.chartConfig.plotReservedSpacePercent) / 100
+    );
+    let spaceUsed = this.componentStore.plot.calculateSpace({
+      width: chartWidth,
+      height: chartHeight,
+    });
+    availableWidth -= spaceUsed.width;
+    availableHeight -= spaceUsed.height;
+
+    spaceUsed = this.componentStore.title.calculateSpace({
+      width: this.chartConfig.width,
+      height: availableHeight,
+    });
+    log.trace('space used by title: ', spaceUsed);
+    plotY = spaceUsed.height;
+    availableHeight -= spaceUsed.height;
+    this.componentStore.xAxis.setAxisPosition('bottom');
+    spaceUsed = this.componentStore.xAxis.calculateSpace({
+      width: availableWidth,
+      height: availableHeight,
+    });
+    log.trace('space used by xaxis: ', spaceUsed);
+    availableHeight -= spaceUsed.height;
+    this.componentStore.yAxis.setAxisPosition('left');
+    spaceUsed = this.componentStore.yAxis.calculateSpace({
+      width: availableWidth,
+      height: availableHeight,
+    });
+    log.trace('space used by yaxis: ', spaceUsed);
+    plotX = spaceUsed.width;
+    availableWidth -= spaceUsed.width;
+    if (availableWidth > 0) {
+      chartWidth += availableWidth;
+      availableWidth = 0;
+    }
+    if (availableHeight > 0) {
+      chartHeight += availableHeight;
+      availableHeight = 0;
+    }
+    this.componentStore.plot.calculateSpace({
+      width: chartWidth,
+      height: chartHeight,
+    });
+
+    log.trace(
+      `Final chart dimansion: x = ${plotX}, y = ${plotY}, width = ${chartWidth}, height = ${chartHeight}`
+    );
```

Comment:
Are we going to leave all `log.trace` inclusions in final solution? cc @sidharthv96

Reply:
removed the unnecessory logs